src/main.cpp

```diff
@@ -2193,7 +2193,11 @@ DEMUX_PACKET* CInputStreamAdaptive::DemuxRead(void)
       memcpy(p->cryptoInfo->kid, pData, 16);
       pData += 16;
       iSize -= (pData - sr->GetSampleData());
+      ReaderCryptoInfo info = sr->GetReaderCryptoInfo();
       p->cryptoInfo->flags = 0;
+      p->cryptoInfo->cryptBlocks = info.m_cryptBlocks;
+      p->cryptoInfo->skipBlocks = info.m_skipBlocks;
+      p->cryptoInfo->mode = info.m_mode;
```

Comment:
here you are assuming that bento4 variables
AP4_CENC_CIPHER_AES_128_CTR/AP4_CENC_CIPHER_AES_128_CBC
are always equal to mediacodec android variables

are we sure that bento4 will always follow this pattern? i wondering what happens if bento4 change values?
do you think it would be better to define our own enum based on android values?

Reply:
yes we're assuming these values stay the same - equally we're relying on the android side that these values don't change to set the crypto mode either:
```
packet.cryptoInfo->mode == CJNIMediaCodec::CRYPTO_MODE_AES_CBC
                ? CJNIMediaCodec::CRYPTO_MODE_AES_CBC
                : CJNIMediaCodec::CRYPTO_MODE_AES_CTR);

```
So no I could never say universally fixed.... I'll define our own enum in samplereader.h